src/main/java/org/folio/processing/mapping/defaultmapper/processor/functions/NormalizationFunction.java

```diff
@@ -422,6 +424,42 @@ public String apply(RuleExecutionContext context) {
     }
   },
 
+  SET_SUBJECT_SOURCE_ID() {
+    private static final String NAME_PARAMETER = "name";
+
+    @Override
+    public String apply(RuleExecutionContext context) {
+      String typeName = context.getRuleParameter().getString(NAME_PARAMETER);
+      List<SubjectSource> subjectSources = context.getMappingParameters().getSubjectSources();
+      if (subjectSources == null || typeName == null) {
+        return STUB_FIELD_TYPE_ID;
+      }
+      return subjectSources.stream()
+        .filter(identifierType -> identifierType.getName().trim().equalsIgnoreCase(typeName))
+        .findFirst()
+        .map(SubjectSource::getId)
+        .orElse(STUB_FIELD_TYPE_ID);
```

Comment:
Is this stub id corresponds to the default subject source?

Reply:
@KaterynaSenchenko Good point! After verifying with PO I changed this logic and set it as empty if there will be NO MATCH.